### PR TITLE
Move descendant_get from Object to Class

### LINKED ISSUE #62: descendant_get belongs on Class not Object
Since descendant_get uses descendants, it should be on Class since descendants is defined on Class.

### PR DESCRIPTION
**Pull request by Fryguy**
This is technically a breaking change, but it really wasn't possible to
call this on Objects previously.  Most callers called this on Class.

Fixes #62

@bdunne Please review.  Since we are considering a major version release, I figured we should get this in now.

### CODE REVIEW

**Comment by Fryguy** [comment]
Marked as WIP as I'd like to wait on merge until we cut a 3.8 before the major version bump

File: spec/core_ext/class/hierarchy_spec.rb [hunk 1]
```diff
@@ -1,6 +1,24 @@
...
+    it "returns a direct descendant" do
+      expect(IO.descendant_get("BasicSocket")).to eq(BasicSocket)
+    end
+
+    it "returns a transitive descendant" do
+      expect(IO.descendant_get("IPSocket")).to eq(IPSocket)
+    end
+
+    it "raises an error if no descendant is found" do
+      expect { IO.descendant_get("Foo") }.to raise_error(ArgumentError, "Foo is not a descendant of IO")
```
Comment (bdunne) [inline spec/core_ext/class/hierarchy_spec.rb]: Should this be testing a constant that exists, but is not a descendant of IO?  Like `"MoreCoreExtensions"`?

Comment (Fryguy) [reply spec/core_ext/class/hierarchy_spec.rb]: Good one...I'll add a test for that.

**Comment by Fryguy** [comment]
unWIP now that 3.8.0 is released.

**Pull request merged**